### ISSUE TITLE
escapeinside erases preceeding text in console output

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#962]
(Original issue 1255 created by aduret on 2016-07-13T08:18:45.633901+00:00)

I'm using Pygments 2.1.3.

In the following example, using the `escapeinside` option for LaTeX output of console input erases everything that is *before* the escaped sequence.

```
#!sh

% cat console.txt
$ ls
@x@foo
bar@y@
baz

% pygmentize -l console -P escapeinside=@@ -f latex console.txt
\begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8}]
\PY{g+gp}{\PYZdl{}} ls
\PY{esc}{x}\PY{g+go}{foo}
\PY{esc}{y}
\PY{g+go}{baz}
\end{Verbatim}

```
See how the `bar` word disappeared?

This behavior does not occur with the `sh` lexer:

```
#!sh
% pygmentize -l sh -P escapeinside=@@ -f latex console.txt
\begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8}]
\PYZdl{} ls
\PY{esc}{x}foo
bar\PY{esc}{y}
baz
\end{Verbatim}
```

This bug was originally spotted while using minted, and reported [here](https://github.com/gpoore/minted/issues/127).

**Comment by Anteru** [comment]
(Original comment by aduret on 2016-07-13T10:08:27.412889+00:00)

This issue seem to be common to "console session" lexers.  Here is the same issue with `pycon`:

```
#!sh
% cat console.txt
>>> 1+2
@x@3
>>> 4+5
9@y@

% pygmentize -l pycon -f latex -P escapeinside=@@ console.txt
\begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8}]
\PY{g+gp}{\PYZgt{}\PYZgt{}\PYZgt{} }\PY{l+m+mi}{1}\PY{o}{+}\PY{l+m+mi}{2}
\PY{esc}{x}\PY{g+go}{3}
\PY{g+gp}{\PYZgt{}\PYZgt{}\PYZgt{} }\PY{l+m+mi}{4}\PY{o}{+}\PY{l+m+mi}{5}
\PY{esc}{y}
\end{Verbatim}
```

The `9` part of the output was lost.

**Comment by muzimuzhi** [comment]
As its duplicate issue #1516 has been fixed since v2.7.2, this issue is fixed as well hence can be closed.

Now the two examples both give expected output:

- In `console` example, `bar` before escaped `y` is now present.
  ```
  #!sh
  % cat console.txt
  $ ls
  @x@foo
  bar@y@
  baz

  % pygmentize -l console -P escapeinside=@@ -f latex console.txt
  \begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8\relax}]
  \PY{g+gp}{\PYZdl{} }ls
  \PY{esc}{x}\PY{g+go}{foo}
  \PY{g+go}{bar}\PY{esc}{y}
  \PY{g+go}{baz}
  \end{Verbatim}
  ```
- In `pycon` example, `9` before escaped `y` is now present too.
  ```
  #!sh
  % cat console.txt
  >>> 1+2
  @x@3
  >>> 4+5
  9@y@

  % pygmentize -l pycon -f latex -P escapeinside=@@ console.txt
  \begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8\relax}]
  \PY{g+gp}{\PYZgt{}\PYZgt{}\PYZgt{} }\PY{l+m+mi}{1}\PY{o}{+}\PY{l+m+mi}{2}
  \PY{esc}{x}\PY{g+go}{3}
  \PY{g+gp}{\PYZgt{}\PYZgt{}\PYZgt{} }\PY{l+m+mi}{4}\PY{o}{+}\PY{l+m+mi}{5}
  \PY{g+go}{9}\PY{esc}{y}
  \end{Verbatim}
  ```

Visual LaTeX output
![image](https://github.com/pygments/pygments/assets/6376638/5e59ab5c-71e1-4ae4-b131-623a17b749bb)

<details><summary>LaTeX example</summary>
<p>

```tex
\documentclass{report}
\usepackage{minted}

\begin{document}
\begin{minted}[escapeinside=@@]{console}
$ ls
@x@foo
bar@y@
baz
\end{minted}

\begin{minted}[escapeinside=@@]{pycon}
>>> 1+2
@x@3
>>> 4+5
9@y@
\end{minted}
\end{document}
```

</p>
</details>

**Comment by birkenfeld** [comment]
Thanks for the heads up!
